source/server/connection_handler_impl.cc

```diff
@@ -69,71 +60,141 @@ void ConnectionHandlerImpl::ActiveListener::removeConnection(ActiveConnection& c
   parent_.num_connections_--;
 }
 
-ConnectionHandlerImpl::ActiveListener::ActiveListener(
-    ConnectionHandlerImpl& parent, Network::ListenSocket& socket,
-    Network::FilterChainFactory& factory, Stats::Scope& scope, uint64_t listener_tag,
-    const Network::ListenerOptions& listener_options)
-    : ActiveListener(
-          parent, parent.dispatcher_.createListener(parent, socket, *this, scope, listener_options),
-          factory, scope, listener_tag) {}
+ConnectionHandlerImpl::ActiveListener::ActiveListener(ConnectionHandlerImpl& parent,
+                                                      Network::ListenerConfig& config)
+    : ActiveListener(parent,
+                     parent.dispatcher_.createListener(config.socket(), *this, config.bindToPort()),
+                     config) {}
 
 ConnectionHandlerImpl::ActiveListener::ActiveListener(ConnectionHandlerImpl& parent,
                                                       Network::ListenerPtr&& listener,
-                                                      Network::FilterChainFactory& factory,
-                                                      Stats::Scope& scope, uint64_t listener_tag)
-    : parent_(parent), factory_(factory), listener_(std::move(listener)),
-      stats_(generateStats(scope)), listener_tag_(listener_tag) {}
+                                                      Network::ListenerConfig& config)
+    : parent_(parent), listener_(std::move(listener)),
+      stats_(generateStats(config.listenerScope())), listener_tag_(config.listenerTag()),
+      config_(config),
+      legacy_stats_(new Filter::Listener::ProxyProtocol::Config(config.listenerScope())) {}
 
 ConnectionHandlerImpl::ActiveListener::~ActiveListener() {
+  while (!sockets_.empty()) {
```

Comment:
these are sockets that are still in filter iteration but have not been converted to connections, right? Can you add some comments?

Reply:
Right, added a comment.